discover/grub2/blscfg.c

```diff
@@ -55,6 +55,25 @@ static char *field_append(struct bls_state *state, int type, char *buffer,
 	return buffer;
 }
 
+static void field_trim(char *buffer)
+{
+        int i, j, end = 0;
+
+        if (!buffer)
+                return;
+
+        for (i = 0; isblank(buffer[i]); i++);
```

Comment:
Keep in mind that `conf_get_pair_space` will be splitting on spaces only, while this will be stripping tabs and spaces. Is that OK?

Reply:
Yes, that's fine. The issue I've described would still be resolved if I just checked `buffer[i] == ' '` instead of `isblank(buffer[i])`, but there's no harm in stripping any errant tabs as well. That was just me trying to make it a little more robust.